clients/src/main/java/org/apache/kafka/clients/consumer/internals/PartitionAssignor.java

```diff
@@ -182,12 +180,16 @@ public ByteBuffer userData() {
             return userData;
         }
 
-        public void setGroupInstanceId(Optional<String> groupInstanceId) {
+        public Optional<String> groupInstanceId() {
+            return groupInstanceId;
+        }
+
+        void setGroupInstanceId(Optional<String> groupInstanceId) {
             this.groupInstanceId = groupInstanceId;
         }
 
-        public Optional<String> groupInstanceId() {
-            return groupInstanceId;
+        Short version() {
```

Comment:
Seems no harm exposing this since it already comes from the assignor? It may even be needed by the assignor to parse a subscription. Similarly in `Assignment`.

Reply:
This version is communicated from assignor to consumer coordinator, i.e. the assignor implementor should always know what's the version right now?